### PR TITLE
Cleaning legacy

### LINKED ISSUE #105: Remove legacy files
**Problem summary/missing feature:**  
The change of name from hac-game-lib to pygamelib required deprecating the hac-game-lib module and that required some extra files. It is time to clean that up.

**Expected behavior:**  
Remove the extra files

**Work to do:**  
Remove the following files:

- [ ] Readme-Legacy.md
- [ ] setup-legacy.py

### LINKED ISSUE #115: [NEW] Update sphinx dependencies
**Problem summary/missing feature:**  
@Frans06 reported the following error when using `make all`:
```bash
sphinx_rtd_theme is no longer a hard dependency since version 1.4.0. Please install it manually.(pip install sphinx_rtd_theme)-g python-sphinx
n
```

**Expected behavior:**  
The theme must be installed with sphinx.

**Work to do:**  
Check the dependencies and update the requirements (pipenv requirements as well as .legacy/ ones).

### PR DESCRIPTION
**Pull request by Frans06**
# Pull Request Template

## Description

Cleaning some Legacy files as well as fixing some dependancies.

Fixes #115 Fixes #105 

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] make all
- [X] Files removed 

**Test Configuration**:
* OS: Linux
* OS version: 5.8.14-arch1-1
* Python version: 3.8
* Architecture: x86_64

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings

### CODE REVIEW

**Review by arnauddupuis — CHANGES_REQUESTED** [review]
This is overall good however the .legacy/requirements.txt (and the mention in the Pipfile) should not be removed.
They are kept because some of the automatic toolings we are using (like read the docs) do not support Pipenv dependencies yet, only requirements.txt files.
So until all the external services (RTD, CircleCI, codecov, etc.) support Pipenv we need to keep that one.

File: setup.py [hunk 1]
```diff
@@ -50,5 +50,4 @@
         "Release Notes": "https://8bitscoding.io/2020/08/30/pygamelib-v1-2-0-release-notes/",  # noqa: E501
     },
     python_requires=">=3.6",
-    obsolotes="hac-game-lib",
```
Comment (arnauddupuis) [inline setup.py:53]: At first, I didn't mind that one, but after more considerations, I think we should keep that. There is no garantuee that people made the switch in just one major release. We should keep the depreciation at least until v1.5.0.

**Review by arnauddupuis — CHANGES_REQUESTED** [review]
Could you please revert your changes to the `obsoletes` parameter?
I explained why after more consideration, we should keep it.

File: setup.py (hunk 1 above)
Comment (Frans06) [reply setup.py:53]: there is a typo there. "obsolotes". that is correct?

Comment (arnauddupuis) [reply setup.py:53]: absolutely... There's a typo... :facepalm:

**Review by arnauddupuis — APPROVED** [review]

**Pull request merged**